cmake/Findamxc.cmake

```
@@ -0,0 +1,34 @@
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+# SPDX-FileCopyrightText: 2019-2020 the prplMesh contributors (see AUTHORS.md)
+#
+# This code is subject to the terms of the BSD+Patent license.
+# See LICENSE file for more details.
+
+find_library(AMXC_LIBRARY "libamxc.so" PATHS ${CMAKE_PREFIX_PATH}/usr/lib/)
+find_path(AMXC_INCLUDE_DIRS
+    NAMES amxc_llist.h amxc_string_join.h amxc_common.h amxc_rbuffer.h amxc_variant_type.h amxc_variant.h amxc_aqueue.h amxc_timestamp.h amxc_astack.h amxc_hash.h amxc_array.h amxc.h amxc_lstack.h amxc_string.h amxc_htable.h amxc_lqueue.h amxc_string_split.h
+    PATH_SUFFIXES amxc
+)
```

You don't need to list all of them, typically `amxc.h` should be enough.

Also, typically you'd include it with `#include <amxc/amxc.h>` so you should search directly for that file.

```suggestion
find_path(AMXC_INCLUDE_DIRS NAMES amxc/amxc.h)
```

---

Applied. Only thing, that `PATH_SUFFIXES` [already point to the subdirectory which should be checked](https://cmake.org/cmake/help/latest/command/find_path.html?highlight=find_path), so i am not sure that is worth to implicitly add `amxc/amxc.h`.